chrome-extension/sidebar/sidebar.js

```diff
@@ -48,6 +48,318 @@ const htmlToElement = (html) => {
+
+    let fetchPromise = fetch(fullURL, {
+      method: 'GET',
+    })
+    .then(response => {
+      if (!response.ok) {
+        courseMostCommonGradeDict[courseName] = 0; // update dictionary value to 0
+        return null;
+      }
+      return response.text();
+    })
+    .then(html => {
+      if (html) {
+        // Get the information we need from the html
+        const parser = new DOMParser();
+        const doc = parser.parseFromString(html, 'text/html');
+        const spans = doc.querySelectorAll('.css-sbue0t');
+        
+        if (spans.length > 0) {
+          const lastSpan = spans[spans.length - 1];
+          let myStr = lastSpan.innerHTML;
+          let myNum = parseFloat(myStr.match(/<!-- -->(\d+)<!-- -->%/)[1]);
+          courseMostCommonGradeDict[courseName] = myNum; // update dictionary to desired value
+          return myNum;
+        } else {
+          return null;
+        }
+      }
+    })
+    .catch(error => {
+      console.log('Error: ' + error);
+    });
+
+    fetchPromises.push(fetchPromise);
+  }
+
+  // Wait for all fetch promises to resolve
+  Promise.all(fetchPromises)
+    .then(() => {
+      // Once all promises have resolved (fetch requests are complete and data is populated), sort courseDivs using same process as before
+      courseDivs.sort((a, b) => {
+        let nameA = a.querySelector('a').getAttribute('name');
+        let nameB = b.querySelector('a').getAttribute('name');
+        return courseMostCommonGradeDict[nameB] - courseMostCommonGradeDict[nameA];
+      });
+
+      console.log(courseMostCommonGradeDict);
+      console.log(courseDivs);
+
+      courseListResults.firstElementChild.innerHTML = '';
+      courseDivs.forEach(courseDiv => {
+          courseListResults.firstElementChild.appendChild(courseDiv);
+      });
+
+      document.querySelector('.size-dropdown').value = 'C';
+    })
+    .catch(error => {
+      console.error('Error waiting for fetch promises:', error);
+    });
+};
+
+// NOTE: the rest of the functions follow very similar code from the last two.
+// The first one is to build dictionaries from information that is already on the schedule builder page (eg. seats available, course units)
+// The second one is to build dictionaries from information using an API (eg. popularity, most-common grade)
+// Refer to them if you are confused about something
+
+const sortByPopularity = () => {
+  let coursePopularityDict = {};
+
+  // Get the main container with class 'course-list-results'
+  let courseListResults = document.querySelector('.course-list-results');
+  let courseDivs = Array.from(courseListResults.firstElementChild.children);
+
+  // Array to hold all fetch promises
+  let fetchPromises = [];
+
+  for (let courseDiv of courseDivs) {
+    let courseName = courseDiv.querySelector('a').getAttribute('name');
+    let fullURL = `https://corsproxy.io/?https://umn.lol/class/${courseName}?static=all`;
```

Comment:
same thing here use the API

Reply:
Same problem with the CORS policy.
![Screenshot 2024-06-20 014400](https://github.com/samyok/gophergrades/assets/72324738/f1dc4cf3-798d-49a5-a7cc-d5766650dc3e)